### ISSUE TITLE
Plugins for native Linux Teamspeak Client (Task)

### ISSUE THREAD

**Issue by ninelore** [IDI-Systems/acre2#1055]
After #653 is practically solved, I think its worth to open a new issue for this specific subject. I'm opening this to collect ideas and requirements for making ACRE work with native Teamspeak clients instead of using a Windows Teamspeak client within the Arma 3 wineprefix.
I'm currently reading myself into the ACRE extension source code and into wine IPC.

My suggestions

- if possible only adjust the pipe location accordingly when a OS condition is met.

It would be nice if you could provide me with following information

- what source files are relevant to the actual pipe creation and access
- if a pipe file is created: its location
- more tba

Resourcces: 
[Wine Server on the WineHQ Wiki](https://wiki.winehq.org/Wine_Developer%27s_Guide/Architecture_Overview#The_Wine_server) (It handles the IPC)

**Comment by jonpas** [comment]
- Pipe implementation
  - https://github.com/IDI-Systems/acre2/tree/master/extensions/src/ACRE2Arma/arma2ts)
- Pipe creation/use
  - https://github.com/IDI-Systems/acre2/blob/master/extensions/src/ACRE2Core/NamedPipeServer.h
  - https://github.com/IDI-Systems/acre2/blob/master/extensions/src/ACRE2Core/NamedPipeServer.cpp
- Pipe use in TeamSpeak plugin
  - https://github.com/IDI-Systems/acre2/blob/f638bba78faa2acd9833d8277b59ee432d011d63/extensions/src/ACRE2TS/TSCallbacks_init.cpp#L12-L13

**Comment by ninelore** [comment]
Thanks for the Infos. Ill ask here if i need anything else. 

Progress can be seen [here](https://github.com/users/ninelore/projects/1)

**Comment by rhysday** [comment]
So it's not possible to get Acre2 in Proton to connect with Teamspeak on Linux?

**Comment by ninelore** [comment]
> So it's not possible to get Acre2 in Proton to connect with Teamspeak on Linux?

It should be possible, but my work on it came to an halt due to be being busy and the wine api being not well documented

**Comment by rhysday** [comment]
yea fair enough. I managed to get Teamspeak installed in the Arma3 Proton container. But just getting teamspeak to be able to launch with Arma3 is the challenge. Seems it might be possible if I create a .bat script in the Proton container.

**Comment by rhysday** [comment]
Hey man I just got TS and ACRE2 working on Linux by following these instructions:
https://ninelore.github.io/projects/#arma-on-linux

I did find TS crashing a few times but will have to run more tests.

**Comment by jonpas** [comment]
Will link https://www.ninelore.org/project/armaonlinux/ on our wiki.

**Comment by rautamiekka** [comment]
> Will link https://www.ninelore.org/project/armaonlinux/ on our wiki.

Not happy, but I looked over the instructions and the `Arma3Helper.sh`, seems valid to work AFAICT.

**Comment by jonpas** [comment]
> Not happy

Clarify?

**Comment by ChaosRifle** [comment]
i can confirm some applications are capable of this. linked is example code for an opentrack version that hooks into wine/proton and effectively tunnels out of it and into the native application, so it can interact. Given you can edit the way you output, it may not be necessary at all, but worst case, this should be feasible. 
https://github.com/Priton-CE/opentrack-StarCitizen

**Comment by DanielGaaA** [comment]
I know this request is abandoned but there could be possibility to make it work using AF_UNIX socket files instead of pipes for IPC. Windows support AF_UNIX socket since 2017.

Wine/proton doesn't support it AF_UNIX socket yet but it is in progress. https://gitlab.winehq.org/wine/wine/-/merge_requests/7650

**Comment by ChaosRifle** [comment]
I have since learned this can also be done using pipes over localhost. SRS for DCS is done this way.
